### PR TITLE
Aggregate all hooks in Before or After super-step fo cucumberjvm-2/3/4

### PR DESCRIPTION
**Pull request by clicman**
- Fold all hooks in Before or After super-steps for cucumberjvm2 (mess with #190\)

The new behavior is to fold em to single steps. It is better than completely disable it.

image

### CODE REVIEW

**Comment by baev** [comment]
can we use Set Up / Tear Down stages instead of having meta steps?

**Comment by baev** [comment]
All you need it create `TestResultContainer` and add `TestResult#uuid` to container's children

**Comment by clicman** [comment]
Okay. I'll look into tomorrow.

File: allure-cucumber2-jvm/src/test/java/io/qameta/allure/cucumber2jvm/AllureCucumber2JvmTest.java [hunk 1]
```diff
@@ -221,10 +221,12 @@ void shouldAddBackgroundSteps() {
                 .flatExtracting(TestResult::getSteps)
                 .extracting(StepResult::getName)
                 .containsExactly(
+                        "Before",
```
Comment (baev) [inline allure-cucumber2-jvm/src/test/java/io/qameta/allure/cucumber2jvm/AllureCucumber2JvmTest.java]: Could you please create separate test for that feature

**Comment by baev** [comment]
Also it would be so nice to add this feature to cucumberjvm3 and cucumberjvm4 as well

**Comment by clicman** [comment]
Let's follow the next steps:
1. Crate acceptable implementation. (You should say what I can proceed to step 2)
2. Port it to cucumber3/4.
3. Write tests.

**Comment by baev** [comment]
BTW Next release scheduled to upcoming Sunday

**Comment by clicman** [comment]
@baev Please take a look at [this branch](https://github.com/allure-framework/allure-java/compare/master...clicman:feature/hook-stages?expand=1). I've tried to implement stages, it looks good in debug structure, but report builds with all steps in test execution stage; Set Up and Tear Down stages are absent in report:
image

Storage structure before wrting container:
image

**Comment by clicman** [comment]
**Migrate to Stages for befores and afters**
**Add testcase complex status logic:**

* Before: failed, Steps: skipped, After: passed = skipped
* Before: failed, Steps: skipped, After: failed = skipped
* Before: passed, Steps: passed, After:  failed = broken
* Before: passed, Steps: falied, After:  passed = falied
* Before: passed, Steps: falied, After:  failed = falied

@baev stage 1 is finished.

**Comment by clicman** [comment]
@baev Tests were pushed. Please review PR and let me know, it's ok or not, to I start to port it into another adaptors.

**Comment by clicman** [comment]
@baev 
- Ported to all adaptors.
- Status logic is unified between adapters.

**Comment by clicman** [comment]
@baev any ETA for release?

**Comment by baev** [comment]
Today

**Comment by symonk** [comment]
I don't work with java anymore, but this can probably close #190

**Comment by baev** [comment]
I want to keep #190 for a while, maybe we will add a possibility to mark fixture as hidden. In that case all the information will be collected but hidden in the report UI

**Comment by clicman** [comment]
@baev does it will be applied in allure.properties for example?

**Comment by baev** [comment]
I am thinking of separate annotation for such purpose and property to set default behavior

**Pull request merged**